File: Swift/ExampleTests/production/OrderItemCollection.swift
public let orderItemCollectionMinimum = 0
public let orderItemCollectionMaximum = 100

public class OrderItemCollection {
    private var items: [OrderItem] = []

    public init() {}

    public func add(_ item: OrderItem) {
        items.append(item)
    }

    public func delete(_ item: OrderItem) -> Bool where OrderItem: Equatable {
        if let idx = items.firstIndex(where: { $0 == item }) {
            items.remove(at: idx)
            return true
        }
        return false
    }

    public func read() -> [OrderItem] {
        return items
    }

    public func update(_ oldItem: OrderItem, with newItem: OrderItem) -> Bool where OrderItem: Equatable {
        if let idx = items.firstIndex(where: { $0 == oldItem }) {
            items[idx] = newItem
            return true
        }
        return false
    }

    public var size: Int { return items.count }
}

File: JavaScript/ExampleTests/production/OrderItemCollection.js
import { OrderItem } from "./OrderItem.js";

export class OrderItemCollection {
  static MINIMUM = 0;
  static MAXIMUM = 100;

  #items = [];

  add(item) { this.#items.push(item); }

  delete(item) {
    const idx = this.#items.indexOf(item);
    if (idx < 0) return false;
    this.#items.splice(idx, 1);
    return true;
  }

  read() { return [...this.#items]; }

  update(oldItem, newItem) {
    const idx = this.#items.indexOf(oldItem);
    if (idx < 0) return false;
    this.#items[idx] = newItem;
    return true;
  }

  size() { return this.#items.length; }
}
